api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirEncounterDaoImpl.java

```diff
@@ -24,14 +26,23 @@
 @Component
 @Setter(AccessLevel.PACKAGE)
 public class FhirEncounterDaoImpl implements FhirEncounterDao {
-	
+
 	@Inject
 	@Named("sessionFactory")
 	private SessionFactory sessionFactory;
-	
+
 	@Override
 	public Encounter getEncounterByUuid(String uuid) {
 		return (Encounter) sessionFactory.getCurrentSession().createCriteria(Encounter.class).add(eq("uuid", uuid))
-		        .uniqueResult();
+				.uniqueResult();
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public List<Encounter> findEncountersByPatientIdentifier(String patientIdentifier) {
+		return sessionFactory.getCurrentSession().createQuery("FROM Encounter e "
```

Comment:
I'd prefer the criteria version of this:

```
return (List<Encounter>) sessionFactory.getCurrentSession().createCriteria(Encounter.class).createAlias("patient", "p")
				.createAlias("p.identifiers", "pi")
				.add(eq("pi.identifier", patientIdentifier))
				.list();
```

My reason for favouring Criteria over HQL or JPQL is that it will make things easier when we get to implementing search across multiple properties.

Reply:
Got it